### PR TITLE
fix[190]: pass ddb record keys through pipe instead of entire new record

### PR DESCRIPTION
**Pull request by JoshuaToth**
*Issue #, if available:*
190

*Description of changes:*
Pass in the DDB keys object using the Eventbridge pipe, then in the main.py of the ECS task, load the DDB record. This will prevent the 8k character limit for the ECS task override which is currently causing syncing issues with large context windows or many files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: backend/embedding/main.py [hunk 1]
```diff
@@ -206,25 +206,42 @@ def main(
     )
 
 
+def get_bot(user_id: str, sk: str, bot_id: str):
```
Comment (statefb) [inline backend/embedding/main.py:209]: Please replace with `repositories.custom_bot.find_private_bot_by_id`

File: backend/embedding/main.py [hunk 2]
```diff
@@ -206,25 +206,42 @@ def main(
...
+    sk = keys["SK"]["S"]
     bot_id = decompose_bot_id(sk)
 
-    pk = new_image["PK"]["S"]
+    pk = keys["PK"]["S"]
     user_id = pk
 
+    new_image = get_bot(user_id, sk, bot_id)
+
+    knowledge = new_image["Knowledge"]
```
Comment (statefb) [inline backend/embedding/main.py:240]: Please fetch from `app.repositories.models.custom_bot.BotModel`

**Review by statefb — COMMENTED** [review]
Thank you for PR! Could you modify and check whether it works as intended?

**Comment by JoshuaToth** [comment]
testing merged changes, will fix lint error when tests are all good.

**Comment by statefb** [comment]
LGTM, thank you for your contribution!
closes #190

**Pull request merged**